### PR TITLE
Add `ports` field to `FluentBitSpec`

### PR DESCRIPTION
**Pull request by euclidgame**

### CODE REVIEW

**Comment by marshtompsxd** [comment]
Could you rebase and push again? This PR has some redundancy with the previous one.

**Comment by euclidgame** [comment]
Some proof statements might be redundant.

File: src/controller_examples/fluent_controller/fluentbit/exec/resource/service.rs [hunk 1]
```diff
@@ -156,4 +164,55 @@ pub fn make_service(fb: &FluentBit) -> (service: Service)
...
+            service_ports@.map_values(|p: ServicePort| p@),
+            Seq::new(i as nat, |k: int| model_resource::make_service_port(ports[k]@))
+        );
+    }
+    while i < ports.len()
+        invariant
+            i <= ports@.len(),
+            service_ports@.map_values(|p: ServicePort| p@) == Seq::new(i as nat, |k: int| model_resource::make_service_port(ports[k]@)),
+        ensures
+            service_ports@.map_values(|p: ServicePort| p@) == Seq::new(ports@.len(), |k: int| model_resource::make_service_port(ports[k]@)),
```
Comment (marshtompsxd) [inline src/controller_examples/fluent_controller/fluentbit/exec/resource/service.rs:183]: I am a bit surprised that you don't need to write `decrease` here. Did I miss anything?

Comment (marshtompsxd) [reply src/controller_examples/fluent_controller/fluentbit/exec/resource/service.rs:183]: Seems that the tutorial does not even have that https://verus-lang.github.io/verus/guide/while.html

Comment (euclidgame) [reply src/controller_examples/fluent_controller/fluentbit/exec/resource/service.rs:183]: Not sure I follow. But I thought `decreases` only exists in spec functions. And there's no recursion.

Comment (marshtompsxd) [reply src/controller_examples/fluent_controller/fluentbit/exec/resource/service.rs:183]: I am referring to this https://dafny.org/latest/DafnyRef/DafnyRef#sec-loop-termination
But I guess Verus does not support that yet

Comment (euclidgame) [reply src/controller_examples/fluent_controller/fluentbit/exec/resource/service.rs:183]: In this case the `i` is decreasing? But in our case `i` is increasing.

Comment (marshtompsxd) [reply src/controller_examples/fluent_controller/fluentbit/exec/resource/service.rs:183]: `ports@.len() - i`

Comment (marshtompsxd) [reply src/controller_examples/fluent_controller/fluentbit/exec/resource/service.rs:183]: You can try to add `decrease` clause to this loop and see what Verus will say

**Pull request merged**